### PR TITLE
Default organization names

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This PR introduces the idea of default organization names. Previously auto-created organizations had no name. Now, we are giving them an auto-generated name when the user first logs in and does not have an orgid so the frontend can display something.

The tricky aspect of this is that we do want to overwrite this name when the user successfully submits a registration - this means that the registration is no longer a "draft", however we don't want to overwrite it if the user has created the organization themselves. To solve this I've introduced a `CreatedBy` field on the organization record.

SC-11446

### Type of change

- [ ] bug fix
- [x] new feature
- [ ] documentation
- [ ] other (describe)

### Acceptance criteria

Does the default organization name format and name overriding logic make sense?

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [x]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: pkg/bff/emails/client.go [hunk 1]
```diff
@@ -76,31 +75,22 @@ func (m *EmailManager) SendUserInvite(user *management.User, inviter *management
...
+	if ctx.Inviter, err = auth.UserDisplayName(inviter); err != nil {
+		return err
 	}
 
-	if userName != "" {
-		ctx.User = userName
-	} else {
-		ctx.User = userEmail
+	if ctx.User, err = auth.UserDisplayName(user); err != nil {
+		return err
```
Comment (pdeziel) [inline pkg/bff/emails/client.go:93]: Should be no functional changes here, just trying to clean up the logic a bit.

File: pkg/bff/users.go [hunk 2]
```diff
@@ -102,9 +102,17 @@ func (s *Server) Login(c *gin.Context) {
...
 		// This is a new user so create a new organization for them
-		org = &models.Organization{}
+		var userName string
+		if userName, err = auth.UserDisplayName(user); err != nil {
+			log.Error().Err(err).Str("user_id", *user.ID).Msg("could not get user display name")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not complete user login"))
+			return
+		}
+		org = &models.Organization{
+			Name: fmt.Sprintf("Registration By %s [Draft]", userName),
```
Comment (pdeziel) [inline pkg/bff/users.go:112]: Does this name format make sense?

**Review by bbengfort — COMMENTED** [review]
Looks like a good start! I had one question and a suggestion for a change that I think will make managing the org names a bit easier.

File: pkg/bff/organization.go [hunk 3]
```diff
@@ -73,8 +73,9 @@ func (s *Server) CreateOrganization(c *gin.Context) {
 
 	// Create a new organization in the database with the provided name and domain
 	org := &models.Organization{
-		Name:   params.Name,
-		Domain: domain,
+		Name:      params.Name,
+		Domain:    domain,
+		CreatedBy: *user.ID,
```
[the remark this answers is not in the document]
Comment (pdeziel) [reply pkg/bff/organization.go]: The nil check happens in the `UserInfo` middleware now.

**Review by bbengfort — APPROVED** [review]

**Pull request merged**